### PR TITLE
Feature/cli refactor

### PR DESCRIPTION
**Pull request by geclos**
# WAT
Refactors CLI a little bit.
- DRY'd some repeated code
- Split some responsibilities
- Added decorators for easy configuration of some common command tasks
- Simplified the spawn syntax
- Improved overall readability
- Added per setup step to the prepare command
- Fixed an issue where CLI was outputting an error message when trying to sync a .env file that does not exist
- [x] Tests

### CODE REVIEW

File: packages/cli/src/commands/update/index.ts [hunk 1]
```diff
@@ -55,22 +58,24 @@ async function getVersions({ fix }: { fix: boolean }) {
...
-// This means user had installed a different version and wants to fix it
+// If --fix flag is passed, use the version defined in latitude.json This means
+// user had installed a different version and wants to fix it
 export default async function updateCommand(args: { fix?: boolean }) {
   const fix = args.fix ?? false
   const { oldVersion, newVersion } = await getVersions({ fix })
 
   // Errors already handled
-  if (!newVersion) return
+  // TODO: process.exit(1)?
```
Comment (andresgutgon) [inline packages/cli/src/commands/update/index.ts:72]: What this todo means?

File: packages/cli/src/commands/build/index.ts [hunk 2]
```diff
@@ -1,94 +1,102 @@
...
+    const targetPath = path.join(config.cwd, 'build')
+    const appPath = path.join(config.cwd, '.latitude', 'app')
+
+    try {
+      const exists = await fs.pathExists(targetPath)
+      if (exists) await fs.remove(targetPath)
+
+      await fs.symlink(appPath, targetPath)
+    } catch (e) {
+      console.log('error: ', e)
```
Comment (andresgutgon) [inline packages/cli/src/commands/build/index.ts:64]: 🔥

**Review by csansoon — DISMISSED** [review]

File: packages/cli/tsconfig.json [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
   "extends": "@latitude-data/typescript/base.json",
   "compilerOptions": {
-    "rootDir": "src",
-    "outDir": "dist"
+    "rootDir": ".",
+    "baseUrl": ".",
+    "outDir": "dist",
+    "paths": {
+      "$src": ["src"],
+      "$src/*": ["src/*"]
```
Comment (andresgutgon) [inline packages/cli/tsconfig.json:9]: both paths are necessary?

Comment (geclos) [reply packages/cli/tsconfig.json:9]: yup

**Pull request merged**